### ISSUE TITLE
[bug] Symbolic links are not properly copied when importing on Linux

### ISSUE THREAD

**Issue by cqjjjzr** [conan-io/conan#6252]
### Environment Details (include every applicable attribute)
  * Operating System+version: Ubuntu 19.10
  * Compiler+version: GCC 9.2.1
  * Conan version: Conan 1.21.0
  * Python version: Python 3.7.5

### Steps to reproduce (Include if Applicable)

`conanfile.txt`:
```
[requires]
expat/2.2.8

[imports]
bin, *.dll -> .
lib, *.dylib -> .
lib, *.so -> .
```

### Actual Behaviour

After using `conan install .`, I got these in my `~/.conan/data/expat/2.2.8/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709/lib`:  
libexpat.so  libexpat.so.1  libexpat.so.1.6.10

And the result of `stat`  is:

```
  File: libexpat.so -> libexpat.so.1
  Size: 13        	Blocks: 0          IO Block: 4096   symbolic link
Device: 802h/2050d	Inode: 666147      Links: 1
File: libexpat.so.1 -> libexpat.so.1.6.10
  Size: 18        	Blocks: 0          IO Block: 4096   symbolic link
Device: 802h/2050d	Inode: 666148      Links: 1
```

When importing, those symbolic links with relative paths are directly copied into project dir, and they became invalid, failing the build process.

### Expected Behaviour

Source file of symbolic links should be copied and imported instead of link itself.

### Logs (Executed commands with output) (Include/Attach if Applicable)
Command: `conan install .`

```
expat/2.2.8: Downloaded package revision 0
conanfile.txt: Generator txt created conanbuildinfo.txt
conanfile.txt: Generated conaninfo.txt
conanfile.txt: Generated graphinfo
conanfile.txt imports(): Copied 1 '.so' file: libexpat.so
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/conans/client/command.py", line 1947, in run
    method(args[0][1:])
  File "/usr/local/lib/python3.7/dist-packages/conans/client/command.py", line 481, in install
    lockfile=args.lockfile)
  File "/usr/local/lib/python3.7/dist-packages/conans/client/conan_api.py", line 81, in wrapper
    return f(api, *args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/conans/client/conan_api.py", line 565, in install
    recorder=recorder)
  File "/usr/local/lib/python3.7/dist-packages/conans/client/manager.py", line 100, in deps_install
    run_imports(conanfile, install_folder)
  File "/usr/local/lib/python3.7/dist-packages/conans/client/importer.py", line 86, in run_imports
    _report_save_manifest(copied_files, import_output, dest_folder, IMPORTS_MANIFESTS)
  File "/usr/local/lib/python3.7/dist-packages/conans/client/importer.py", line 61, in _report_save_manifest
    file_dict[f] = md5sum(abs_path)
  File "/usr/local/lib/python3.7/dist-packages/conans/util/files.py", line 136, in md5sum
    return _generic_algorithm_sum(file_path, "md5")
  File "/usr/local/lib/python3.7/dist-packages/conans/util/files.py", line 149, in _generic_algorithm_sum
    with open(file_path, 'rb') as fh:
FileNotFoundError: [Errno 2] No such file or directory: '/home/charliejiang/temp/libexpat.so'
```

**Comment by jgsogo** [comment]
Hi! In this case you are not copying all the needed files, you need to write `lib, *.so* -> .` in order to copy the `libexpat.so` and the file/s it is linking to. If all the files are in place, Conan won't fail there and it will successfully create the `conan_imports_manifest.txt` file.

Thanks for this issue, we will improve the error and provide a meaningful message. 👍

**Comment by cqjjjzr** [comment]
Maybe it's a good idea not to copy the whole .so file and keep a symlink instead? This avoids unnecessary copying.

**Comment by cen1** [comment]
Hah, I jut got bitten by this. I think it is more than enough to simply add this into imports documentation and a small "warning" section for linux. It is easy to overlook and the error is cryptic, hard to track down with google even.

**Comment by buffge** [comment]
the demo should to be lib, \*.so\* -> ./bin

**Comment by Renari** [comment]
I think the underlying issue is that inside the `imports` block of a conanfile the `preserve symlinks` option isn't available.

**Comment by kambala-decapitator** [comment]
also faced this in v1.58.0 with `*.so` pattern, "strange" crash with message pointing to the actually copied symlink is not something you would expect.

**Comment by memsharded** [comment]
Using ``imports`` is not the recommended way any longer, as it has been dropped in Conan 2.0.

The recommended way is using the ``generate()`` method, iterating the ``self.dependencies`` to access the dependencies information, and using the new ``copy()`` (``from conan.tools.file import copy``, not the ``self.copy``, as it does way less things with symlinks). Also, if ``copy()`` is still not enough, it is always possible to do raw ``shutil.copy()`` or whatever other file manipulation that needs to be done, as the folders like the origin ``self.dependencies["mydep"].cpp_info.bindir`` are available in the ``generate()`` method.

I am closing this as wont fix (2.0 is due this month), please report against the new ``copy()`` tools. Thanks!
